### DISCUSSION TITLE
我想确认一下新版本许可证的限制

### DISCUSSION THREAD

**Discussion by Tianjie-Wu** [dashingsoft/pyarmor#1976] [General]
您好，

我有两个开发环境，离线的开发环境我已经购买了集团版许可证，我阅读了相关文档，只需要重新生成设备注册即可，没有问题。

我的在线开发环境用的是专业版许可证，会在featurize平台上（租显卡的）调试程序并加密。我比较担心很容易超过新许可证的100个设备的限制。根据我的理解，每次开发时启动的容器会算成一个设备，每天会使用1-2次新容器，因此是否意味着我的专业版许可证两个月左右就会无法继续使用？还是说新版本许可证和旧许可证一样，依旧是每天100个设备的限制而不是总限制？麻烦您解答一下我的疑惑，谢谢！

**Comment by jondy** [comment]
```
在新版本 9 中，专家版许可证 1 个月可以运行 100 次，超过 100 次之后，必须到下一个月才可以使用

准确的说，每运行一次会占用一个 token, 这个 token 在 30 天之后才会释放，一个许可证最多可以有 100 个 token
```
